lib/chef/knife/cookbook_site_install.rb

```diff
@@ -67,12 +67,24 @@ class CookbookSiteInstall < Knife
         :default => "https://supermarket.chef.io",
         :proc => Proc.new { |supermarket| Chef::Config[:knife][:supermarket_site] = supermarket }
 
+      option :no_vendor,
+        :long => "--no-vendor",
+        :description => "Do not commit cookbook to a git vendor branch",
+        :boolean => true,
+        :default => false
+
       attr_reader :cookbook_name
       attr_reader :vendor_path
 
       def run
         extend Chef::Mixin::ShellOut
 
+        # Deprecation warning for vendor installs.
+        unless config[:no_vendor]
+          Chef.deprecated(:cookbook_install_vendor, "Using knife cookbook site install with vendor branching is deprecated. " \
+            "Please pass --no-vendor or add `knife[:no_vendor] = true` to your knife configuration.")
```

Comment:
Is the idea that in Chef 15 we just remove this option and make it the default? If so we might want to mention that here.